internal/log/slog.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+// NewLoggerSlog returns a *slog.Logger that prints in the provided format at the
+// provided level with a UTC timestamp and the caller of the log entry.
+func NewLoggerSlog(c Config) (*slog.Logger, error) {
+	lvlOption, err := parseLevel(c.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	logger, err := parseFmt(lvlOption, c.Format)
+	if err != nil {
+		return nil, err
+	}
+
+	return logger, nil
+}
+
+func replaceSlogAttributes(_ []string, a slog.Attr) slog.Attr {
```

Comment:
I'm wondering why do we want to replace this attributes?

Reply:
If you initialize the [other log structure](https://github.com/prometheus-operator/prometheus-operator/blob/main/internal/log/log.go#L55) it's possible to see that it had different attributes. The same ones that I'm putting in this new implementation

The steps that I took to discover it were:

```Go
func TestWhyWeNeed2Replace(t *testing.T) {
        // initialize the NewLogger from internal package
	t.Log("TestTmp")
	l, err := NewLogger(Config{
		Level:  LevelDebug,
		Format: FormatJSON,
	})
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	level.Info(l).Log("msg", "test")

	// initialize a default slog.Logger
	l2 := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
		Level:     slog.LevelDebug,
		AddSource: true}))

	l2.Info("test")
}
```

Output NewLogger:

```JSON
{"caller":"slog_test.go:166","level":"info","msg":"test","ts":"2024-07-13T11:08:18.119068Z"}
```

Output default Slog(without changed the attributes)

```JSON
{"time":"2024-07-13T08:13:35.356893-03:00","level":"INFO","source":{"function":"github.com/prometheus-operator/prometheus-operator/internal/log.TestWhyWeNeed2Replace","file":"/Users/jonathan.silva/prometheus-operator/internal/log/slog_test.go","line":175},"msg":"test"}
```

In that case, it's possible to notice that the attributes that I changed are the ones that are different between them: 

- `time -> ts`
- `level string in lower case`
- `source -> caller`

Reply:
I got, but why do we want to replace it?
Why not use the default info provided by slog?
Maybe I missing some part, sorry if it sounds weird hehe 😄

Reply:
> Why do we want to replace it?

I thought that was an agreement to keep both implementations using the same patterns... If isn't, who do I need to talk to decide it?

Reply:
but if you are ok to no replace it, that is it...

Reply:
I mean change it might or might not be considered a breaking change.

WHat's your opinion @simonpasquier @ArthurSens @slashpai ?

Reply:
I think we can move forward with no breaking changes in the log attributes for now. If we decide this is too much work to main, we can change it later!